dom.go

```diff
@@ -101,30 +109,38 @@ type Node struct {
 
 // HTML returns the HTML representation of the node.
 func (e Node) HTML() template.HTML {
+	return template.HTML(e.buildHTML(&strings.Builder{}).String())
+}
+
+func (e Node) buildHTML(sb *strings.Builder) *strings.Builder {
 	if e.Name == "" {
```

Comment:
gist of the change actually: while `HTML()` public interface remains unchanged, we use a new internal helper function `buildHTML(sb *strings.Builder) *strings.Builder` so that the entire generating of HTML string is done with a single `strings.Builder` to be efficient